### PR TITLE
fix #11958 [BUG] python generates wrong model name and model file name

### LINKED ISSUE #11958: [BUG] python generates wrong model name and model file name
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

oas.yaml ( notice with multiple `allOf` and with a CJK title):

```yaml
openapi: 3.0.2
info:
  title: foo
  version: v1
  description: 'foo'
servers:
  - url: 'foo.com'
    description: 'foo'
tags:
  - name: foo
    description: 'foo'

paths:
  
  '/foos':
    get:
      summary: foo list
      operationId: foo-list
      tags:
        - foo
      responses:
        '200':
          description: good
 
components:

  schemas:
  
    Bar:
      allOf:
      - type: object
        title: 中文
        properties:
          data:
            type: object
      - type: object
        title: 中文
        properties:
          data2:
            type: object
```

config.yaml:

```yaml
projectName: foo
packageName: foo
packageVersion: "1.0"
```

##### openapi-generator version

5.4.0

##### OpenAPI declaration file content or url

see above

##### Steps to reproduce

```bash
openapi-generator-cli  generate \
  --generator-name python \
  --config config.yaml \
  --output output \
  --input-spec oas.yaml 
```

got model file:

```
├── model
│   ├── 1.py
│   ├── __init__.py
│   └── bar.py
```

Look at file `1.py`, should be `model_1.py`.

It's content also has problem:

```python
class 1(ModelNormal):
```

Not a valid class name.

### PR DESCRIPTION
**Pull request by chanjarster**
fix #11958 [BUG] python generates wrong model name and model file name

Modify AbstractPythonCodegen.toModelName just like AbstractJavaCodegen.toModelName

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@taxpon @frol @mbohlool @cbornet @kenjones-cisco @tomplus @arun-nalla @spacether

### CODE REVIEW

**Comment by chanjarster** [comment]
@wing328 anybody can review this PR?

**Comment by spacether** [comment]
Hey there. Thank you for this PR.
Can you include a test that verifies that this is working correctly?
Otherwise someone could remove or break this functionality in the future and we would have no signal letting us know that this is working/not working correctly.
One could add tests in this file:
- https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java
How about checking:
- modelNamePrefix is used
- modelNameSuffix is used
- model name is camelized
- a reserved model name is renamed (important)
- a model name starting with a digit is renamed (important)

**Comment by chanjarster** [comment]
@spacether unit test added,

**Comment by spacether** [comment]
Thank you for adding the unit test. Your python tests are breaking in node4, can you fix them?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java [hunk 1]
```diff
@@ -655,14 +655,14 @@ public String toModelName(String name) {
 
         // model name cannot use reserved keyword, e.g. return
         if (isReservedWord(camelizedName)) {
-            String modelName = "model_" + camelizedName; // e.g. return => ModelReturn (after camelize)
+            String modelName = "Model_" + camelizedName; // e.g. return => ModelReturn (after camelize)
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java:658]: What do you think about removing the underscore? It's not a normal character for class names in python.

Comment (chanjarster) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java:658]: You're right.

**Comment by chanjarster** [comment]
updated, all check passed

**Review by spacether — APPROVED** [review]
This looks great. Thank you for your PR!

**Pull request merged**